### PR TITLE
fix solo12_hardware_calibration

### PR DESCRIPTION
**Pull request by luator**
## Description

Two fixes:
1. `count` needs to be incremented in the loop (also reduce the print rate a bit).
2. The while loop after calling `request_calibration()` was not executed as `robot_ready` was still `true` from before requesting calibration. Avoid that by changing to a do-while loop.  Also move the sleep to the beginning, as it needs a moment for the reported state to be updated to non-ready after `request_calibration()`.

## How I Tested

By running it on the test bench.

## I fulfilled the following requirements

[//]: # "Please make sure you followed these steps before requesting a review."
[//]: # "Check the boxes in the list below, when done."

- [x] All new code is formatted according to our style guide (for C++ run clang-format, for Python, run flake8 and fix all warnings).
- [x] All new functions/classes are documented and existing documentation is updated according to changes.
- [x] No commented code from testing/debugging is kept (unless there is a good reason to keep it).

### CODE REVIEW

File: src/programs/solo12_hardware_calibration.cpp [hunk 1]
```diff
@@ -89,13 +90,17 @@ int main(int argc, char** argv)
     shared_content.sc_mutex.unlock();
 
     // print the home offset:
-    while (!robot_ready)
+    do
```
Comment (MaximilienNaveau) [inline src/programs/solo12_hardware_calibration.cpp:93]: here you could have while loop before to "wait until ready", I guess the solo have that.

**Review by MaximilienNaveau — CHANGES_REQUESTED** [review]
Just a minor comment

File: src/programs/solo12_hardware_calibration.cpp (hunk 1 above)
Comment (luator) [reply src/programs/solo12_hardware_calibration.cpp:93]: I can give it a try.  In this case we would be missing the lock around the "is ready" check, though.  But I'm not sure if this is actually needed here.

**Review by MaximilienNaveau — APPROVED** [review]
My bad it's correct

**Pull request merged**